BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -368,12 +368,25 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
           next
         end
 
-        self.addlog("Saving result file #{path}")
-        name = File.basename(path)
-        
+        # Get name and filetype
+        self.addlog("Attempting to save result file #{path}")
+        name = File.basename(path)         
         userfile_class = Userfile.suggested_file_type(name)
-        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile )
+        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile ) 
+       
+        # If the file already exists, add a tag (task id + ns time) to the name
+        if userfile_exists(userfile_class, :name => name)
+          tag = '-' + (self.id.to_s || 'unk') + '-' + Time.now.nsec.to_s
```

Comment:
I agree; that is what I originally did, but I thought it would be a bit safer against race conditions.